src/OpenLoco/src/Ui/Chart.cpp

```diff
@@ -0,0 +1,298 @@
+#include "Chart.h"
+#include "Economy/Currency.h"
+#include "Graphics/SoftwareDrawingEngine.h"
+#include "Graphics/TextRenderer.h"
+#include "Ui/Window.h"
+
+namespace OpenLoco::Ui
+{
+    static uint32_t _dword_113658C; // 0x0113658C
+
+    // TODO: replace with templated functions
+    static int64_t graphGetValueFromPointer(const std::byte* dataPtr, const uint8_t dataTypeSize)
+    {
+        switch (dataTypeSize)
+        {
+            case 2:
+                return *reinterpret_cast<const int16_t*>(dataPtr);
+
+            case 4:
+                return *reinterpret_cast<const int32_t*>(dataPtr);
+
+            case 6:
+                return reinterpret_cast<const currency48_t*>(dataPtr)->asInt64();
+
+            default:
+                return 0;
+        }
+    }
+
+    // 0x004CF869
+    static int64_t graphGetMaxValue(const GraphSettings& gs)
+    {
+        int64_t maxValue = 0;
+        for (auto lineIndex = 0U; lineIndex < gs.lineCount; lineIndex++)
+        {
+            auto dataIndex = 0U;
+            std::byte* dataPtr = gs.yData[lineIndex];
+            if ((gs.flags & GraphFlags::dataFrontToBack) != GraphFlags::none)
+            {
+                dataIndex = gs.dataStart[lineIndex];
+                dataPtr = &dataPtr[gs.dataTypeSize * (gs.dataEnd - dataIndex)];
+            }
+
+            while (dataIndex < gs.dataEnd)
+            {
+                // Data front-to-back?
+                // NB: all charts except cargo delivery
+                if ((gs.flags & GraphFlags::dataFrontToBack) != GraphFlags::none)
+                {
+                    dataPtr -= gs.dataTypeSize;
+                }
+
+                int64_t value = graphGetValueFromPointer(dataPtr, gs.dataTypeSize);
+                maxValue = std::max(maxValue, std::abs(value));
+                dataIndex++;
+
+                // Data back-to-front?
+                // NB: for cargo delivery chart
+                if ((gs.flags & GraphFlags::dataFrontToBack) == GraphFlags::none)
+                {
+                    dataPtr += gs.dataTypeSize;
+                }
+            }
+        }
+
+        return maxValue;
+    }
+
+    // 0x004CFA49
+    static void graphDrawAxesAndLabels(const GraphSettings& gs, Window* self, Gfx::DrawingContext& drawingCtx)
+    {
+        auto xAxisLabelValue = gs.xAxisRange;
+        if ((gs.flags & GraphFlags::dataFrontToBack) != GraphFlags::none)
+        {
+            xAxisLabelValue -= (gs.dataEnd - 1) * gs.xAxisStepSize;
+        }
+
+        // 0x004CFA74
+        for (auto ecx = 0U; ecx < gs.dataEnd; ecx++)
+        {
+            auto remainder = xAxisLabelValue % gs.xAxisLabelIncrement;
+
+            // Draw vertical lines for each of the data points
+            {
+                auto xPos = ecx * gs.word_113DD80 + gs.left + gs.xOffset;
```

Comment:
do we know enough to name word_113DD80

Reply:
We already have `yAxisLabelIncrement`, so this would be `xAxisLabelIncrement`.